packages/image_picker/image_picker/ios/Classes/FLTImagePickerPlugin.m

```diff
@@ -67,32 +117,26 @@ - (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result
   }
 
   if ([@"pickImage" isEqualToString:call.method]) {
-    _imagePickerController = [[UIImagePickerController alloc] init];
-    _imagePickerController.modalPresentationStyle = UIModalPresentationCurrentContext;
-    _imagePickerController.delegate = self;
-    _imagePickerController.mediaTypes = @[ (NSString *)kUTTypeImage ];
-
     self.result = result;
     _arguments = call.arguments;
-
     int imageSource = [[_arguments objectForKey:@"source"] intValue];
 
-    switch (imageSource) {
-      case SOURCE_CAMERA: {
-        NSInteger cameraDevice = [[_arguments objectForKey:@"cameraDevice"] intValue];
-        _device = (cameraDevice == 1) ? UIImagePickerControllerCameraDeviceFront
-                                      : UIImagePickerControllerCameraDeviceRear;
-        [self checkCameraAuthorization];
-        break;
+    if (imageSource == SOURCE_GALLERY) {  // Capture is not possible with PHPicker
+      if (@available(iOS 14, *)) {
+        // PHPicker is used
+        [self pickImageWithPHPicker:true];
+      } else {
+        // UIImagePicker is used
+        [self pickImageWithUIImagePicker];
       }
-      case SOURCE_GALLERY:
-        [self checkPhotoAuthorization];
-        break;
-      default:
-        result([FlutterError errorWithCode:@"invalid_source"
-                                   message:@"Invalid image source."
-                                   details:nil]);
-        break;
+    } else {
+      [self pickImageWithUIImagePicker];
+    }
```

Comment:
nits, these code can be shorten to:
```objc
if (imageSource == SOURCE_GALLERY && @available(iOS 14, *)) { 
// PHPicker is only available for picking photos from gallery and with iOS 14 and above.
   [self pickImageWithPHPicker:true];
 } else {
   [self pickImageWithUIImagePicker];
 }

```

Reply:
I tried it but it gives this warning.
> @available does not guard availability here; use if (@available) instead

I think we should use `@available(iOS 14, *)` alone. What do you think?

Reply:
Ah I see. The current code is good then.